pkg/scheduler/framework/plugins/nodevolumelimits/csi.go

```diff
@@ -80,6 +81,35 @@ func (pl *CSILimits) EventsToRegister() []framework.ClusterEvent {
 	}
 }
 
+// PreFilter invoked at the prefilter extension point
+//
+// If the pod only use secrets and configmaps, we'll skip the Filter phase
+func (pl *CSILimits) PreFilter(ctx context.Context, _ *framework.CycleState, pod *v1.Pod) (*framework.PreFilterResult, *framework.Status) {
+	// If the new pod doesn't have any volume attached to it, then we'll skip the Filter phase
```

Comment:
Nit: do we need this for loop here?

The cost of calling for "_, vol := range pod.Spec.Volumes {" for empty pod.Spec.Volumes should be negligible or zero and having this if here makes the code slightly more complicated.